modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -408,23 +408,24 @@ INSFVMomentumAdvection::coeffCalculator(const Elem & elem) const
     mooseAssert((neighbor == &fi->elem()) || (neighbor == fi->neighborPtr()),
                 "Surely the neighbor has to match one of the face information's elements, right?");
 
-    ADRealVectorValue neighbor_velocity(_u_var->getNeighborValue(neighbor, *fi, elem_velocity(0)));
+    // Checking if skewness correction is necesary
+    bool correct_skewness =
+        (_u_var->faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage);
+
+    // Fill a face velocity for the advection contribution
+    ADRealVectorValue face_velocity(
+        _u_var->getInternalFaceValue(neighbor, *fi, elem_velocity(0), correct_skewness));
     if (_v_var)
```

Comment:
should we check that u, v and w all request skewness correction ?
I reckon in some meshes (extruded) it could make sense to skew-correct only some directions but feels pretty niche

Reply:
Great idea! The latest commit checks the skewness-parameter for every direction separately.